flake.nix

```diff
@@ -1,69 +1,80 @@
 {
-  # This is a template created by `hix init`
   inputs = {
-    nixpkgs = {
-      follows = "std/nixpkgs";
-    };
-    hackage = {
-      url = "github:input-output-hk/hackage.nix";
-      flake = false;
-    };
-    haskellNix = {
-      inputs = {
-        hackage.follows = "hackage";
-      };
-      url = "github:input-output-hk/haskell.nix";
-    };
-    std = {
-      url = "github:divnix/std";
-    };
-    iohk-nix = {
-      url = "github:input-output-hk/iohk-nix";
-      inputs = {
-        nixpkgs.follows = "nixpkgs";
-      };
-    };
+    haskellNix.url = "github:input-output-hk/haskell.nix";
+    nixpkgs.follows = "haskellNix/nixpkgs-unstable";
+    iohkNix.url = "github:input-output-hk/iohk-nix";
+    flake-utils.url = "github:hamishmack/flake-utils/hkm/nested-hydraJobs";
+
+    CHaP.url = "github:input-output-hk/cardano-haskell-packages?ref=repo";
+    CHaP.flake = false;
+
+    # non-flake nix compatibility
+    flake-compat.url = "github:edolstra/flake-compat";
+    flake-compat.flake = false;
   };
-  outputs = {
-    std,
-    self,
-    ...
-  } @ inputs:
-    std.growOn {
-      inherit inputs;
-      systems = [
+
+  outputs = inputs:
+    let
+      profiling = false;
+      supportedSystems = [
         "x86_64-linux"
         "x86_64-darwin"
+        # not supported on ci.iog.io right now
+        #"aarch64-linux"
         "aarch64-darwin"
-      ];
-      cellsFrom = ./cells;
-      cellBlocks = [
-        (std.functions "library")
-        (std.installables "packages")
-        (std.runnables "apps")
-        (std.nixago "nixago")
-        (std.functions "devshellProfiles")
-        (std.devshells "devshells")
-        (std.installables "ciJobs")
-        (std.functions "actions")
-      ];
-    }
-    # soil layers:
-    {
-      # nix-cli compat
-      packages = std.harvest self [["prelude" "packages"] ["prelude" "library"] ["_automation" "ciJobs"] ["_automation" "packages"]];
-      devShells = std.harvest self ["_automation" "devshells"];
+       ]; in
+    let flake = inputs.flake-utils.lib.eachSystem supportedSystems (system:
+      let
+        # setup our nixpkgs with the haskell.nix overlays, and the iohk-nix
+        # overlays...
+        nixpkgs = import inputs.nixpkgs {
```

Comment:
Often I take  the short cut and use directly `haskellNix.legacyPackages.${system}`, which already comes with overlays and stuff (since `nixpkgs` is already `haskellNix/nixpkgs-unstable`).